plugins/package-managers/python/src/main/kotlin/Uv.kt

```diff
@@ -0,0 +1,138 @@
+import org.ossreviewtoolkit.plugins.packagemanagers.python.utils.getPythonVersion
+import org.ossreviewtoolkit.plugins.packagemanagers.python.utils.getPythonVersionConstraint
+import org.ossreviewtoolkit.plugins.packagemanagers.python.utils.toOrtPackages
+import org.ossreviewtoolkit.plugins.packagemanagers.python.utils.toPackageReferences
+import org.ossreviewtoolkit.utils.common.CommandLineTool
+import org.ossreviewtoolkit.utils.common.safeDeleteRecursively
+import org.ossreviewtoolkit.utils.ort.createOrtTempFile
+
+internal object UvCommand : CommandLineTool {
+    override fun command(workingDir: File?) = "uv"
+
+    override fun transformVersion(output: String) = output.substringAfter("version ").removeSuffix(")")
+}
+
+/**
+ * [Uv](https://github.com/astral-sh/uv) package manager for Python.
+ */
+@OrtPlugin(
+    displayName = "Uv",
+    description = "An extremely fast Python package and project manager.",
+    factory = PackageManagerFactory::class
+)
+class Uv(override val descriptor: PluginDescriptor = UvFactory.descriptor, private val config: PipConfig) :
+    PackageManager("Uv") {
+    companion object {
+        /**
+         * The name of the build system requirements and information file used by modern Python packages.
+         */
+        internal const val PYPROJECT_FILENAME = "pyproject.toml"
+    }
+
+    override val globsForDefinitionFiles = listOf("uv.lock")
+
+    override fun resolveDependencies(
+        analysisRoot: File,
+        definitionFile: File,
+        excludes: Excludes,
+        analyzerConfig: AnalyzerConfiguration,
+        labels: Map<String, String>
+    ): List<ProjectAnalyzerResult> {
+        val scopeName = parseScopeNamesFromPyproject(definitionFile.resolveSibling(PYPROJECT_FILENAME))
+        val resultsForScopeName = scopeName.associateWith { inspectLockfile(definitionFile) }
+
+        val packages = resultsForScopeName
+            .flatMap { (_, results) -> results.packages }
+            .toOrtPackages()
+            .distinctBy { it.id }
+            .toSet()
+
+        val project = Project.EMPTY.copy(
+            id = Identifier(
+                type = projectType,
+                namespace = "",
+                name = definitionFile.relativeTo(analysisRoot).path,
+                version = VersionControlSystem.getCloneInfo(definitionFile.parentFile).revision
+            ),
+            definitionFilePath = VersionControlSystem.getPathInfo(definitionFile).path,
+            scopeDependencies = resultsForScopeName.mapTo(mutableSetOf()) { (scopeName, results) ->
+                Scope(scopeName, results.resolvedDependenciesGraph.toPackageReferences())
+            },
+            vcsProcessed = processProjectVcs(definitionFile.parentFile)
+        )
+
+        return listOf(ProjectAnalyzerResult(project, packages))
+    }
+
+    /**
+     * Return the result of running Python inspector against a requirements file generated by exporting the dependencies
```

Comment:
Oh, so you *are* using `python-inspector` after all... not that falling back to using `requirements.txt` files should really just be a last resort, as those files only contain a flat list of direct dependencies without much metadata.

I was under the assumption that a modern manager like `uv` offers better ways to query dependencies.

Reply:
We could make a native onw, but since python-inspector is getting back on track, i did as same way.